### PR TITLE
Redirect unit tests logs to file

### PR DESCRIPTION
**Pull request by merlimat**
### Motivation

We are currently disabling the logs for the unit tests execution. The primary reason for that is to avoid logging excessively and reach the max amount of output for Travis. 

The problem with this approach is that when tests are failing we cannot see the logs.

Since we're moving to Jenkins builds, a better option is to enable INFO logs on all the tests and then configure maven to redirect the tests output to files. 

The output for each test will be available in the test-result view and in the Jenkins workspace.

### Modifications

Make sure all the modules are printing info logs and redirect them to files

### CODE REVIEW

File: pulsar-broker/src/test/resources/log4j.properties [hunk 1]
```diff
@@ -1,41 +0,0 @@
...
-# under the License.
-#
-
-
-#
-# Pulsar-Broker Tests Logging Configuration
-#
-# Format is "<default threshold> (, <appender>)+
-
-log4j.rootLogger=OFF, CONSOLE
```
Comment (rdhabalia) [inline pulsar-broker/src/test/resources/log4j.properties]: with this file we could control logging level while running test. after removing this file, can we control logging level while running tests?

Comment (merlimat) [inline pulsar-broker/src/test/resources/log4j.properties]: I couldn't find a way to have a single file to control all the modules. 

With this patch the level can be controlled with system properties. Take a look at the top level `pom.xml` where in the surefire config we can have: 

```
 -Dorg.slf4j.simpleLogger.log.org.apache.bookkeeper.mledger=info
```

Comment (rdhabalia) [inline pulsar-broker/src/test/resources/log4j.properties]: I see..ok

File: pom.xml [hunk 2]
```diff
@@ -557,10 +596,17 @@ flexible messaging model and an intuitive client API.</description>
...
         <artifactId>maven-surefire-plugin</artifactId>
-        <version>2.19.1</version>
         <configuration>
           <argLine> -Xmx2G -XX:MaxDirectMemorySize=8G
-            -Dio.netty.leakDetectionLevel=advanced</argLine>
+            -Dio.netty.leakDetectionLevel=advanced
+            -Dorg.slf4j.simpleLogger.showDateTime=true
+            -Dorg.slf4j.simpleLogger.log.org.apache.zookeeper=off
+            -Dorg.slf4j.simpleLogger.log.org.apache.bookkeeper=off
+            -Dorg.slf4j.simpleLogger.log.org.apache.bookkeeper.mledger=info
```
Comment (saandrews) [inline pom.xml]: Where is the file location mentioned?

**Pull request merged**